postgres/datadog_checks/postgres/postgres.py

```diff
@@ -382,46 +382,60 @@ def _query_scope(self, cursor, scope, instance_tags, is_custom_metrics, relation
             )
             results = results[:MAX_CUSTOM_RESULTS]
 
-        desc = scope['descriptors']
+        # A descriptor is the association of a Postgres column name (e.g. 'schemaname')
+        # to a tag name (e.g. 'schema').
+        descriptors = scope['descriptors']
+
+        # Parse and submit results.
+
+        num_results = 0
 
-        # parse & submit results
-        # A row should look like this
-        # (descriptor, descriptor, ..., value, value, value, value, ...)
-        # with descriptor a PG relation or index name, which we use to create the tags
-        valid_results_size = 0
         for row in results:
-            # Check that all columns will be processed
-            assert len(row) == len(cols) + len(desc)
+            # A row contains descriptor values on the left (used for tagging), and
+            # metric values on the right (used as values for metrics).
+            # E.g.: (descriptor, descriptor, ..., value, value, value, value, ...)
+
+            expected_number_of_columns = len(descriptors) + len(cols)
+            if len(row) != expected_number_of_columns:
+                raise RuntimeError(
+                    'Row does not contain enough values: '
+                    'expected {} ({} descriptors + {} columns), got {}'.format(
+                        expected_number_of_columns, len(descriptors), len(cols), len(row)
+                    )
+                )
+
+            descriptor_values, column_values = row[: len(descriptors)], row[len(descriptors) :]
```

Comment:
Easier to read if written as two lines?

```suggestion
            descriptor_values = row[: len(descriptors)]
            column_values = row[len(descriptors) :]
```

Reply:
Sure!